src/zope/app/generations/browser/managerdetails.py

```diff
@@ -73,15 +150,16 @@ def getEvolvers(self):
 
         for gen in range(manager.minimum_generation, manager.generation):
 
-            info = manager.getInfo(gen+1)
+            info = manager.getInfo(gen + 1)
             if info is None:
                 info = ''
             else:
                 # XXX: the renderer *expects* unicode as input encoding (ajung)
-                renderer = ReStructuredTextToHTMLRenderer(
-                    unicode(info), self.request)
+                renderer = _ReStructuredTextToHTMLRenderer(
+                    info.decode('utf-8') if isinstance(info, bytes) else info,
+                    self.request)
```

Comment:
I think you can remove that `XXX` comment now that you handle both bytes and unicode.